homeassistant/components/zeroconf/__init__.py

```diff
@@ -379,11 +381,38 @@ async def async_setup(self) -> None:
             self.zeroconf, types, handlers=[self.async_service_update]
         )
 
+        async_dispatcher_connect(
+            self.hass,
+            config_entries.SIGNAL_CONFIG_ENTRY_CHANGED,
+            self._handle_config_entry_changed,
+        )
+
     async def async_stop(self) -> None:
         """Cancel the service browser and stop processing the queue."""
         if self.async_service_browser:
             await self.async_service_browser.async_cancel()
 
+    @callback
+    def _handle_config_entry_changed(
+        self,
+        change: config_entries.ConfigEntryChange,
+        entry: config_entries.ConfigEntry,
+    ) -> None:
+        """Handle config entry changes."""
+        if (
+            change != config_entries.ConfigEntryChange.REMOVED
+            or entry.source != config_entries.SOURCE_IGNORE
```

Comment:
Should we trigger rediscovery also when non-ignored config entries are removed for a more consistent experience, i.e. removing any discoverable config entry will make it show up again as discovered?

Reply:
Sounds good.